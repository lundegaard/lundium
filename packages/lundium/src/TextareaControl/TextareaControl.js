import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import FormControl from '../FormControl';

const TextareaControl = ({ children, className, type = 'text', ...rest }) => (
	<FormControl
		className={cx(className)}
		component="textarea"
		type={type}
		{...rest}
	>
		{children}
	</FormControl>
);

TextareaControl.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	/** HTML type attribute of the input. */
	type: PropTypes.string,
};

export default TextareaControl;
