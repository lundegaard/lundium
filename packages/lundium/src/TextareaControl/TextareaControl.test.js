import React from 'react';
import { shallow } from 'enzyme';

import TextareaControl from './TextareaControl';

describe('<TextareaControl />', () => {
	const requiredProps = {
		name: 'textarea-control',
	};

	it('shallow', () => {
		const wrapper = shallow(<TextareaControl {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
