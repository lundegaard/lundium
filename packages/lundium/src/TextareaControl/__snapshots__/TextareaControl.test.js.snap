// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextareaControl /> shallow 1`] = `
<FormControl
  className=""
  component="textarea"
  name="textarea-control"
  type="text"
/>
`;
